infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/receive/LogReplicationSinkManager.java

```diff
@@ -395,36 +411,33 @@ private boolean isValidSnapshotStart(LogReplication.LogReplicationEntryMsg entry
      *
      * @param entry a SNAPSHOT_START message
      */
-    private void processSnapshotStart(LogReplication.LogReplicationEntryMsg entry) {
-        long topologyId = entry.getMetadata().getTopologyConfigID();
-        long timestamp = entry.getMetadata().getSnapshotTimestamp();
+    private void processSnapshotStart(LogReplicationEntryMsg entry) {
+        baseSnapshotTimestamp = entry.getMetadata().getSnapshotTimestamp();
 
         // Signal start of snapshot sync to the writer, so data can be cleared (on old snapshot syncs)
-        snapshotWriter.reset(topologyId, timestamp);
+        snapshotWriter.reset(entry.getMetadata().getTopologyConfigID(), baseSnapshotTimestamp);
 
-        // Update lastTransferDone with the new snapshot transfer timestamp.
-        baseSnapshotTimestamp = entry.getMetadata().getSnapshotTimestamp();
-
-        // Setup buffer manager.
+        // TODO V2: If the sink crashes in the middle of a snapshot transfer cycle, a SNAPSHOT_START message
+        //  won't come and the buffer will not get created, resulting in an NPE
```

Comment:
If sink crashes at this point, then the rest of snapshot_sync msgs, whatever allowed window of pendingAck msgs on source is, will get dropped because the `rxState` is set after creation of the buffer on L425.
Then Source will continue sending the same set of msgs until ACK is received.

So I think we are safe from NPE.